sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+                tags: TestDataGenerator.GenerateTags());
+            var createdDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                 resourceGroupName: resourceGroupName,
+                 dnsResolverName: dnsResolverName,
+                 parameters: dnsResolver);
+            var updatedDnsResolverTags = TestDataGenerator.GenerateTags(startFrom: dnsResolver.Tags.Count);
+            dnsResolver.Tags = updatedDnsResolverTags;
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: dnsResolver);
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.Tags.Should().NotBeNull().And.BeEquivalentTo(updatedDnsResolverTags);
+        }
+
+
+        [Fact]
+        public void PutDnsResolver_ResolverExists_ExpectConflicts()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var existingDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName);
+            var dnsResolverForCreation = TestDataGenerator.GenerateDnsResolver(location: Constants.DnsResolverLocation, subscriptionId: this.SubscriptionId, resourceGroupName: resourceGroupName);
+            
+            Action createDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: existingDnsResolver.Name,
+                parameters: dnsResolverForCreation);
+
+            createDnsResolverAction.Should().NotBeNull();
+            createDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.Conflict);
+        }
+ 
+        [Fact]
+        public void PutDnsResolver_ResolverNotExistsIfNoneMatchSuccess_ExpectResolvereCreated()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolverForCreation = TestDataGenerator.GenerateDnsResolver(location: Constants.DnsResolverLocation,  subscriptionId: this.SubscriptionId, resourceGroupName: resourceGroupName);
+            var ifNoneMatch = "*";
+
+            var createdDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName,
+                parameters: dnsResolverForCreation,
+                ifNoneMatch: ifNoneMatch);
+
+            createdDnsResolver.Should().NotBeNull();
+            createdDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded.ToString());
+        }
+
+        [Fact]
+        public void GetDnsResolver_ResolverExistsByResolverName_ExpectResolverRetrieved() 
```

Comment:
We need to check the response for content, not just provisioning state.

Reply:
`BeSuccessfullyCreated` validates the Dns Resolver is successfully provisioned. 
`BeSameAsExpected` validates  Dns Resolver is same as parameters.

Reply:
`BeSuccessfullyCreated` validates the Dns Resolver is successfully provisioned. 
`BeSameAsExpected` validates  Dns Resolver is same as parameters.